ipn/local.go

```diff
@@ -431,56 +445,105 @@ func (b *LocalBackend) updateFilter(netMap *controlclient.NetworkMap) {
 		b.e.SetFilter(filter.NewAllowNone(b.logf))
 		return
 	}
+	var data struct {
```

Comment:
maybe something more descriptive than `data`? Maybe just use 3 variables, and make UpdateHash take a `...interface{}` and then hash the whole `[]interface{}` ?